include/odp/api/spec/pool.h

```diff
@@ -290,10 +290,16 @@ odp_pool_t odp_pool_lookup(const char *name);
 /**
  * Pool information struct
  * Used to get information about a pool.
+ * @note The difference between end_addr & start_addr
+ * will result in buffers address range belong to this pool.
  */
 typedef struct odp_pool_info_t {
 	const char *name;          /**< pool name */
 	odp_pool_param_t params;   /**< pool parameters */
+	/** Minimum address of any packet contained in this pool */
+	uintptr_t start_addr;
+	/** Maximum address of any packet contained in this pool */
+	uintptr_t end_addr;
```

Comment:
I'd change these to "visible address" rather than "address". The point is that the spec says nothing about how packets are represented within an implementation. It only states what applications may see by using other ODP APIs that manipulate odp_packet_t object.

Reply:
The only other ambiguity is whether the address that applications see for a packet is constant. VPP clearly assumes this but this is not implied by the ODP spec. The "lifetime" of packet visibility is from the time a call like `odp_packet_data()` is made until the thread releases the `odp_packet_t` that was used to obtain this address. If some other thread receives that handle and calls `odp_packet_data()` there's no requirement or guarantee that it will get the same address as the previous owner.

Reply:
1.  Agreed for first suggestion.
2.  For Second, As per my understanding, the VPP only needs to know the range of addresses in order to calculate OFFSET within. VPP doesn't assume/store packet address beyond its lifetime (Rx to Tx)

Reply:
I will wait for other's comments, if any, before sending V3 patchset

Reply:
This spec should apply for all pool types (also other than packet). 0 address may be used when there's no data in an event (e.g. timeout).

/** Minimum data address.
   * This is the minimum address that application accessible data of any 
   * object (event) allocated from the pool may locate. When there's no 
   * application accessible data (e.g. ODP_POOL_TIMEOUT pools), the 
   * value maybe zero. */
uintptr_t min_data_addr;

/** Maximum data address.
   * This is the maximum address that application accessible data of any 
   * object (event) allocated from the pool may locate. When there's no 
   * application accessible data (e.g. ODP_POOL_TIMEOUT pools), the 
   * value maybe zero.  */
uintptr_t max_data_addr;

Reply:
Looks much better. Thank you @psavol for your suggestions.